libs/src/services/contracts/ContractClient.js

```diff
@@ -13,11 +13,12 @@ const METHOD_CALL_MAX_RETRIES = 5
  * time a method on the contract is invoked.
  */
 class ContractClient {
-  constructor (web3Manager, contractABI, contractRegistryKey, getRegistryAddress, contractAddress = null) {
+  constructor (web3Manager, contractABI, contractRegistryKey, getRegistryAddress, logger = console, contractAddress = null) {
     this.web3Manager = web3Manager
     this.contractABI = contractABI
     this.contractRegistryKey = contractRegistryKey
     this.getRegistryAddress = getRegistryAddress
+    this.logger = logger || console
```

Comment:
technically someone could supply `null` as an argument to the `ContractClient` constructor and it would break things

<img width="898" alt="Screen Shot 2022-05-02 at 6 34 03 PM" src="https://user-images.githubusercontent.com/17736112/166337601-169652b6-0abc-4310-9dc9-f1fa330177fb.png">

Gotta love dynamically typed languages

Reply:
fair point - we are vulnerable to this same issue pretty much everywhere else so don't think it matters
but yeah nbd, your call

Reply:
Sounds like an issue that should be fixed when we move this section to typescript 👍 I'll stick to being consistent